ee/api/ee_tagged_item.py

```diff
@@ -0,0 +1,29 @@
+from django.db.models import Q
+from rest_framework import serializers
+
+from posthog.models import Tag
+
+
+class EnterpriseTaggedItemSerializerMixin(serializers.Serializer):
+    """
+    Serializer that interacts with EnterpriseTaggedItem model
+    """
+
+    def get_tags(self, obj):
+        return list(obj.tags.values_list("tag__name", flat=True)) if obj.tags else []
+
+    def set_tags(self, tags, obj):
+        if not obj:
+            # If the object hasn't been created yet, this method will be called again on the create method.
+            return
+
+        # Create new tags
+        for tag in tags:
+            tag_instance, _ = Tag.objects.get_or_create(name=tag, team_id=obj.team_id)
+            obj.tags.get_or_create(tag_id=tag_instance.id)
+
+        # Delete tags that are missing
+        obj.tags.exclude(tag__name__in=tags).delete()
+
+        # Cleanup tags that aren't used by team
+        Tag.objects.filter(Q(team_id=obj.team_id) & Q(taggeditems__isnull=True)).delete()
```

Comment:
I wonder if this is something we want to run every time we save a flag/action/insight/etc. Will it be run always? Am I prematurely optimising?

Reply:
Yea, the other alternative was writing a cron job that did this but it seemed like overkill for now. This can definitely be refactored out into a proper cleanup cron, but I think this is fine for now, since I assume writes << read.